core/src/main/java/org/radargun/config/InitHelper.java

```diff
@@ -1,20 +1,46 @@
 package org.radargun.config;
 
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+
 import java.lang.reflect.Method;
+import java.util.Arrays;
 import java.util.Stack;
 
 /**
  * @author Radim Vansa &lt;rvansa@redhat.com&gt;
  */
 public class InitHelper {
+
+   private static final Log log = LogFactory.getLog(InitHelper.class);
+
    public static void init(Object target) {
+      processAnnotatedMethods(target, Init.class);
+   }
+
+   public static void destroy(Object target) {
+      processAnnotatedMethods(target, Destroy.class);
+   }
+
+   private static void processAnnotatedMethods(Object target, Class annotationClass) {
```

Comment:
This is not correct - destroy methods must be called in inverse order than inits!